src/DOM/File/Blob.purs

```diff
@@ -0,0 +1,22 @@
+module DOM.File.Blob
+  ( type_
+  , size
+  ) where
+
+import Prelude ((==))
+import Data.Maybe (Maybe(..))
+
+import DOM.File.Types (Blob())
+
+foreign import typeImpl :: Blob -> String
+
+-- | A string indicating the MIME type of the data contained in the `Blob`.
+-- | Returns `Nothing` if the type is unknown.
+type_ :: Blob -> Maybe String
+type_ blob = let blobType = typeImpl blob
+             in if blobType == ""
+                then Nothing
+                else Just blobType
```

Comment:
I stared at this one a while and wasn't too sure... but yeah, maybe we just expose the string even if it's empty. I guess ideally this should use a type other than string anyway, but `MimeType` just exists in `affjax` at the moment, I've not broken out that stuff yet.